### ISSUE TITLE
system_clock::timepoint explosion example is not supported by the current API

### ISSUE THREAD

**Issue by pauluap** [HowardHinnant/date#185]
The example at 

https://github.com/HowardHinnant/date/wiki/Examples-and-Recipes#components_to_time_point

does not compile because the time_of_day object does not have the hours(), minutes(), and seconds() members.  Is this intentional?

The approach that I came up with is 

```
using namespace date;
auto time = std::chrono::system_clock::now();
auto daypoint = floor<days>(time);
auto ymd = year_month_day(daypoint);   // calendar date
auto tod = time - daypoint;

// Obtain individual components as integers
auto y   = int(ymd.year());
auto m   = unsigned(ymd.month());
auto d   = unsigned(ymd.day());
auto hours = std::chrono::duration_cast<std::chrono::hours>(tod);
auto minutes = std::chrono::duration_cast<std::chrono::minutes>(tod - hours);
auto seconds = std::chrono::duration_cast<std::chrono::seconds>(tod - hours - minutes);
auto h = hours.count();
auto min = minutes.count();
auto s = seconds.count();
```

Is there a better approach or library support that I missed?

**Comment by HowardHinnant** [comment]
I'm not immediately replicating your symptoms.  The existing example code compiles for me.  However I suspect the problem may be use of a templated `time_of_day` where lookup doesn't go into base classes.

Your workaround looks correct.  But I should instead fix `time_of_day`.

**Comment by pauluap** [comment]
Hmm,

VS2013 actually does compile okay.  My everyday work is done in Vim with YouCompleteMe running clang in the background.  I just tried to run the clang command that YCM generates on the command line.  I've attached the output, but it seems that my variant of Clang is having problems before it even hits time_of_day, which are shown as the last three errors in the output.

[time_of_day_output.txt](https://github.com/HowardHinnant/date/files/1166707/time_of_day_output.txt)

clang version 3.9.0 (tags/RELEASE_390/final)
Target: i686-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Program Files (x86)\LLVM\bin

I'm not sure if it's a clang thing or an oddity of clang on windows.  When I get access to a Linux system, I'll see what I can get to happen.

**Comment by HowardHinnant** [comment]
Ok, that's very helpful, thanks.  This environment is using VS-12's implementation of `<chrono>`, which does not support `constexpr`.  The fix is to go to lines 71-99 of date.h and configure the macros to match the VS-2015 and earlier definitions:
```
//   before VS2017
#    define CONSTDATA const
#    define CONSTCD11
#    define CONSTCD14
#    define NOEXCEPT _NOEXCEPT
```
I'm guessing this may be as simple as removing the `&& !defined(__clang__)` from line 71.

**Comment by pauluap** [comment]
Allright, that did fix it for me.  Though, that filter for __clang__ seems pretty deliberate to me.

I traced the change to 55087c6cbdf156a69c57fd9935c567eefc4232a7.  It's not obvious to me why it was added.  

It's certainly harmful to me, running a version of clang built with the MSVC toolchain, but there must be a reason why it was added.

If we wanted to reduce the impact surface, we could do:

```#if defined(_MSC_VER) && (!defined(__clang__) || (_MSC_VER < 1910))```
